<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

    <html xmlns="http://www.w3.org/1999/xhtml">
  
  


  

  <head>
    <title>
      SatelliteBigData (diff) – BioHackathon 2009
    </title>
        <link rel="search" href="../search.1.html" />
        <link rel="help" href="TracGuide.html" />
        <link rel="up" href="SatelliteBigData@action=history.html" title="Page history" />
        <link rel="next" href="SatelliteBigData@action=diff&amp;version=27.html" title="Version 27" />
        <link rel="start" href="../wiki.1.html" />
        <link rel="stylesheet" href="../chrome/common/css/trac.css" type="text/css" /><link rel="stylesheet" href="../chrome/common/css/wiki.css" type="text/css" /><link rel="stylesheet" href="../chrome/common/css/diff.css" type="text/css" />
        <link rel="prev" href="SatelliteBigData@action=diff&amp;version=25.html" title="Version 25" />
        <link rel="shortcut icon" href="../chrome/common/trac.ico" type="image/x-icon" />
        <link rel="icon" href="../chrome/common/trac.ico" type="image/x-icon" />
      <link type="application/opensearchdescription+xml" rel="search" href="../search/opensearch" title="Search BioHackathon 2009" />
    <script type="text/javascript" src="../chrome/common/js/jquery.js"></script><script type="text/javascript" src="../chrome/common/js/trac.js"></script><script type="text/javascript" src="../chrome/common/js/search.js"></script><script type="text/javascript" src="http://hackathon2.dbcls.jp/chrome/common/js/diff.js"></script>
    <!--[if lt IE 7]>
    <script type="text/javascript" src="/chrome/common/js/ie_pre7_hacks.js"></script>
    <![endif]-->
    <link rel="stylesheet" type="text/css" href="../chrome/common/css/diff.css" /><meta name="ROBOTS" content="NOINDEX, NOFOLLOW" />
  </head>
  <body>
    <div id="banner">
      <div id="header">
        <a id="logo" href="../index.html"><img src="../chrome/site/bh2009-logo.jpg" alt="" /></a>
      </div>
      <form id="search" action="../search.1.html" method="get">
        <div>
          <label for="proj-search">Search:</label>
          <input type="text" id="proj-search" name="q" size="18" value="" />
          <input type="submit" value="Search" />
        </div>
      </form>
      <div id="metanav" class="nav">
    <ul>
      <li class="first"><a href="http://hackathon2.dbcls.jp/login">Login</a></li><li><a href="TracGuide.html">Help/Guide</a></li><li><a href="../about.html">About Trac</a></li><li class="last"><a href="../prefs.html">Preferences</a></li>
    </ul>
  </div>
    </div>
    <div id="mainnav" class="nav">
    <ul>
      <li class="first active"><a href="../wiki.1.html">Wiki</a></li><li><a href="../timeline.html">Timeline</a></li><li><a href="../milestone/index.html">Roadmap</a></li><li><a href="../browser.html">Browse Source</a></li><li><a href="../report.html">View Tickets</a></li><li class="last"><a href="../search.1.html">Search</a></li>
    </ul>
  </div>
    <div id="main">
      <div id="ctxtnav" class="nav">
        <h2>Context Navigation</h2>
          <ul>
              <li class="first"><span>&larr; <a class="prev" href="SatelliteBigData@action=diff&amp;version=25.html" title="Version 25">Previous Change</a></span></li><li><a href="SatelliteBigData@action=history.html" title="Page history">Wiki History</a></li><li class="last"><span><a class="next" href="SatelliteBigData@action=diff&amp;version=27.html" title="Version 27">Next Change</a> &rarr;</span></li>
          </ul>
        <hr />
      </div>
    <div id="content" class="wiki">
      <h1>Changes
        between
          <a href="SatelliteBigData@version=25.html">Version 25</a> and
        <a href="SatelliteBigData@version=26.html">Version 26</a> of
        <a href="SatelliteBigData@version=26.html">SatelliteBigData</a>
      </h1>
      <form method="post" id="prefs" action="SatelliteBigData@version=26.html"><div><input type="hidden" name="__FORM_TOKEN" value="2c93bff27ce469855e39283a" /></div>
        <div>
          <input type="hidden" name="action" value="diff" />
          <input type="hidden" name="version" value="26" />
          <input type="hidden" name="old_version" value="25" />
    <label for="style">View differences</label>
    <select id="style" name="style">
      <option selected="selected" value="inline">inline</option>
      <option value="sidebyside">side by side</option>
    </select>
    <div class="field">
      Show <input type="text" name="contextlines" id="contextlines" size="2" maxlength="3" value="2" />
      <label for="contextlines">lines around each change</label>
    </div>
    <fieldset id="ignore">
      <legend>Ignore:</legend>
      <div class="field">
        <input type="checkbox" id="ignoreblanklines" name="ignoreblanklines" />
        <label for="ignoreblanklines">Blank lines</label>
      </div>
      <div class="field">
        <input type="checkbox" id="ignorecase" name="ignorecase" />
        <label for="ignorecase">Case changes</label>
      </div>
      <div class="field">
        <input type="checkbox" id="ignorewhitespace" name="ignorewhitespace" />
        <label for="ignorewhitespace">White space changes</label>
      </div>
    </fieldset>
    <div class="buttons">
      <input type="submit" name="update" value="Update" />
    </div>
        </div>
      </form>
      <dl id="overview">
        <dt class="property time">Timestamp:</dt>
        <dd class="time">
          2009/03/19 17:55:42 (<a class="timeline" href="../timeline@from=2009-03-19T17%253A55%253A42+0900&amp;precision=second.html" title="2009-03-19T17:55:42+0900 in Timeline">16 years</a> ago)
        </dd>
        <dt class="property author">Author:</dt>
        <dd class="author">
          severin
        </dd>
        <dt class="property message">Comment:</dt>
        <dd class="message">
          
          <p>
--
</p>

        </dd>
      </dl>
      <div class="diff">
        <div class="legend" id="diff-legend">
          <h3>Legend:</h3>
          <dl>
            <dt class="unmod"></dt><dd>Unmodified</dd>
            <dt class="add"></dt><dd>Added</dd>
            <dt class="rem"></dt><dd>Removed</dd>
            <dt class="mod"></dt><dd>Modified</dd>
          </dl>
        </div>
        <div class="diff">
  <ul class="entries">
      <li class="entry">
        <h2 id="file0">
          <a href="SatelliteBigData@version=26.html">SatelliteBigData</a>
        </h2>
        <table class="inline" summary="Differences" cellspacing="0">
              <colgroup><col class="lineno" /><col class="lineno" /><col class="content" /></colgroup>
              <thead>
                <tr>
                  <th title="Version 25">
                    <a href="SatelliteBigData@version=25.html#L40">
                      v25</a>
                  </th>
                  <th title="Version 26">
                    <a href="SatelliteBigData@version=26.html#L40">
                      v26</a>
                  </th>
                  <th> </th>
                </tr>
              </thead>
            <tbody class="unmod">
                  <tr>
                          <th>40</th><th>40</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>41</th><th>41</th><td class="l"><span>Data storage will not be a problem (if you have enogh money). But downloading and/or manipulating Big sequence data must be problem for both provider (us=ddbj) and user. Is there a possible solution? And I have a big question: Do you really need and/or use "Short Read Archive?" -- yaskaz</span> </td>
                  </tr>
            </tbody><tbody class="add">
                  <tr class="first">
                          <th> </th><th>42</th><td class="r"><ins>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</ins> </td>
                  </tr><tr>
                          <th> </th><th>43</th><td class="r"><ins>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</ins> </td>
                  </tr><tr class="last">
                          <th> </th><th>44</th><td class="r"><ins>== Discussion ==</ins> </td>
                  </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>42</th><th>45</th><td class="l"><span></span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>43</th><th> </th><td class="l"><span>== Work goup discussions ==</span> </td>
                      </tr>
                      <tr>
                        <th> </th><th>46</th><td class="r"><span>The purpose of this workshop was to discuss the approaches people use to handle their data. We could obviously not go into the very technical approaches as are being applied to very large datasets such as the Short Read Archive because that requires a different expertise.</span> </td>
                      </tr><tr>
                        <th> </th><th>47</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>48</th><td class="r"><span>Three different levels in the data handling pipeline were discussed.</span> </td>
                      </tr><tr>
                        <th> </th><th>49</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>50</th><td class="r"><span>==== Storage ====</span> </td>
                      </tr><tr>
                        <th> </th><th>51</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>52</th><td class="r"><span>We listed the different types of storage solution used in different fields according to the amount and type of data. A distinction was made between storing a lot of small objects versus storing a few huge objects. Issues like latency and variety of data influence this distinction as well.</span> </td>
                      </tr><tr>
                        <th> </th><th>53</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>54</th><td class="r"><span>Protein-protein interaction datasets typically consist of a relatively small number of small objects. This type of data requires no advanced storage systems or tweaks to common systems; a simple RDBMS will do.</span> </td>
                      </tr><tr>
                        <th> </th><th>55</th><td class="r"><span>Data like genome sequences and assembly data involve larger objects (1 genome sequence -&gt; 3Gb), but are still easily manageable on a standard filesystem.</span> </td>
                      </tr><tr>
                        <th> </th><th>56</th><td class="r"><span>Really big objects such as the data from simulations [IS THIS CORRECT?] require specialized storage systems such as ZFS, lustre or PVFS.</span> </td>
                      </tr><tr>
                        <th> </th><th>57</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>58</th><td class="r"><span>In contrast to the above, diffraction results, microarray results or next-gen sequencing reads involve a largish number of objects which become more difficult to query. They are typically still stored in RDBMS but might require some tweaking that digresses from a normalized relational database model. Apart from obvious things to do such as creating good indices, further optimization can be found by using as few joins as possible and therefore organizing the data so that it can be stored in 2 or 3 tables (e.g. eeDB).</span> </td>
                      </tr><tr>
                        <th> </th><th>59</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>60</th><td class="r"><span>Several attendees are looking into new ways to store their data because they are hitting the ceiling of their storage capacity. Several technologies were mentioned, including OGSADAI which is a grid-based solution where you can setup several databases that then can be queried as one. Other technologies like the cloud might also provide part of a solution. Amazon S3 and GoogleBase allow for storing very large amounts of data (large numbers of large objects) and are relatively cheap. The upload is however very slow to these systems which must be taken into account. In addition, these services are commercial and it might be dangerous to store non-public data there. A possible solution mentioned involves creating an encrypted data image to be uploaded instead of the original dataset. Another issue is that these companies might in the future decide to increase their prices or even stop their activities. Using the cloud for data storage therefore means that you still need a local backup in case this happens.</span> </td>
                      </tr><tr>
                        <th> </th><th>61</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>62</th><td class="r"><span>[Jessica: can you add your vertical red line to this as well?]</span> </td>
                      </tr><tr>
                        <th> </th><th>63</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>64</th><td class="r"><span>==== Querying ====</span> </td>
                      </tr><tr>
                        <th> </th><th>65</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>66</th><td class="r"><span>Several approaches to make the data accessible to others were discussed. For smaller datasets regular SQL can be used to get to the data (example: the Ensembl MySQL server on ensembldb.ensembl.org). If the database schema has to be tweaked to allow larger datasets (e.g. using a minimal number of tables such as eeDB) an API becomes necessary. A RESTful API where an object can be retrieved by URL was mentioned as useful in this case. For even larger datasets you want to try and limit the types of queries people can run. This way you can build a toolkit to ask this limited number of questions, but optimize that toolkit so that latency is reduced.</span> </td>
                      </tr><tr>
                        <th> </th><th>67</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>68</th><td class="r"><span>Streaming</span> </td>
                      </tr><tr>
                        <th> </th><th>69</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>70</th><td class="r"><span>&nbsp; SQL -&gt; API -&gt; toolkit</span> </td>
                      </tr><tr>
                        <th> </th><th>71</th><td class="r"><span>&nbsp; &nbsp;^&nbsp; &nbsp; &nbsp; |</span> </td>
                      </tr><tr>
                        <th> </th><th>72</th><td class="r"><span>&nbsp; &nbsp;|&nbsp; &nbsp; &nbsp; |</span> </td>
                      </tr><tr>
                        <th> </th><th>73</th><td class="r"><span>&nbsp; &nbsp;+------+</span> </td>
                      </tr><tr>
                        <th> </th><th>74</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>75</th><td class="r"><span>==== Processing ====</span> </td>
                      </tr><tr>
                        <th> </th><th>76</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>77</th><td class="r"><span>Apart from storage and getting the data out again we briefly discussed how to actually work with these data, i.e. running scripts on them. It is clear that shipment of large datasets should be avoided; it is often much simpler to bring the software to the data instead.</span> </td>
                      </tr><tr>
                        <th> </th><th>78</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>79</th><td class="r"><span>The main message from this discussion was that each database should be optimized for its own data and will therefore be completely different from the next. However, it is the API that can provide interoperability if we can come up with a common set of terms for querying. We believe that current web-technologies like mashups based on very simple APIs like Flickr's can be an alternative to other grid-based approached for integrating data.</span> </td>
                      </tr><tr>
                        <th> </th><th>80</th><td class="r"><span></span> </td>
                      </tr><tr class="last">
                        <th> </th><th>81</th><td class="r"><span>== Work goup discussion topics ==</span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>44</th><th>82</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>45</th><th>83</th><td class="l"><span>next gen sequencer WG (big data and seq analysis)</span> </td>
                  </tr>
            </tbody>
              <tbody class="skipped">
                <tr>
                  <th><a href="SatelliteBigData@version=25.html#L92">…</a></th>
                  <th><a href="SatelliteBigData@version=26.html#L130">…</a></th>
                  <td> </td>
                </tr>
              </tbody>
            <tbody class="unmod">
                  <tr>
                          <th>92</th><th>130</th><td class="l"><span>&nbsp;- NEED data providers to create complete meta-data descriptions.</span> </td>
                  </tr><tr>
                          <th>93</th><th>131</th><td class="l"><span></span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>94</th><th> </th><td class="l"><span>IN THE END the goal is to find biology<del>&nbsp;and getting access to the individual data elements is critical and this can not be just locked away in</del> files that can not be internally accessed.</span> </td>
                      </tr>
                      <tr class="last">
                        <th> </th><th>132</th><td class="r"><span>IN THE END the goal is to find biology<ins>. Having access to the individual data elements is critical and this can not be just locked away inside</ins> files that can not be internally accessed.</span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>95</th><th>133</th><td class="l"><span></span> </td>
                  </tr>
            </tbody><tbody class="rem">
                  <tr class="last first">
                          <th>96</th><th> </th><td class="l"><del>== Results ==</del> </td>
                  </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>97</th><th>134</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>98</th><th>135</th><td class="l"><span>== TODOs ==</span> </td>
                  </tr>
            </tbody>
        </table>
      </li>
  </ul>
</div>
      </div>
  </div>
    </div>
    <div id="footer" lang="en" xml:lang="en"><hr />
      <a id="tracpowered" href="http://trac.edgewall.org/"><img src="../chrome/common/trac_logo_mini.png" height="30" width="107" alt="Trac Powered" /></a>
      <p class="left">
        Powered by <a href="../about.html"><strong>Trac 0.11.6</strong></a><br />
        By <a href="http://www.edgewall.org/">Edgewall Software</a>.
      </p>
      <p class="right">Visit the Trac open source project at<br /><a href="http://trac.edgewall.org/">http://trac.edgewall.org/</a></p>
    </div>
  </body>
</html>
