<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
  
  


  

  <head>
    <title>
      SatelliteBigData – BioHackathon 2009
    </title>
        <link rel="search" href="../search.1.html" />
        <link rel="help" href="TracGuide.html" />
        <link rel="alternate" href="http://hackathon2.dbcls.jp/wiki/SatelliteBigData?version=28&amp;format=txt" type="text/x-trac-wiki" title="Plain Text" />
        <link rel="up" href="SatelliteBigData.html" title="View latest version" />
        <link rel="next" href="SatelliteBigData@version=29.html" title="Version 29" />
        <link rel="start" href="../wiki.1.html" />
        <link rel="stylesheet" href="../chrome/common/css/trac.css" type="text/css" /><link rel="stylesheet" href="../chrome/common/css/wiki.css" type="text/css" />
        <link rel="prev" href="SatelliteBigData@version=27.html" title="Version 27" />
        <link rel="shortcut icon" href="../chrome/common/trac.ico" type="image/x-icon" />
        <link rel="icon" href="../chrome/common/trac.ico" type="image/x-icon" />
      <link type="application/opensearchdescription+xml" rel="search" href="../search/opensearch" title="Search BioHackathon 2009" />
    <script type="text/javascript" src="../chrome/common/js/jquery.js"></script><script type="text/javascript" src="../chrome/common/js/trac.js"></script><script type="text/javascript" src="../chrome/common/js/search.js"></script>
    <!--[if lt IE 7]>
    <script type="text/javascript" src="/chrome/common/js/ie_pre7_hacks.js"></script>
    <![endif]-->
    <meta name="ROBOTS" content="NOINDEX, NOFOLLOW" /><script type="text/javascript">
      jQuery(document).ready(function($) {
        $("#content").find("h1,h2,h3,h4,h5,h6").addAnchor("Link to this section");
      });
    </script>
  </head>
  <body>
    <div id="banner">
      <div id="header">
        <a id="logo" href="../index.html"><img src="../chrome/site/bh2009-logo.jpg" alt="" /></a>
      </div>
      <form id="search" action="../search.1.html" method="get">
        <div>
          <label for="proj-search">Search:</label>
          <input type="text" id="proj-search" name="q" size="18" value="" />
          <input type="submit" value="Search" />
        </div>
      </form>
      <div id="metanav" class="nav">
    <ul>
      <li class="first"><a href="http://hackathon2.dbcls.jp/login">Login</a></li><li><a href="TracGuide.html">Help/Guide</a></li><li><a href="../about.html">About Trac</a></li><li class="last"><a href="../prefs.html">Preferences</a></li>
    </ul>
  </div>
    </div>
    <div id="mainnav" class="nav">
    <ul>
      <li class="first active"><a href="../wiki.1.html">Wiki</a></li><li><a href="../timeline.html">Timeline</a></li><li><a href="../milestone/index.html">Roadmap</a></li><li><a href="../browser.html">Browse Source</a></li><li><a href="../report.html">View Tickets</a></li><li class="last"><a href="../search.1.html">Search</a></li>
    </ul>
  </div>
    <div id="main">
      <div id="ctxtnav" class="nav">
        <h2>Context Navigation</h2>
          <ul>
              <li class="first"><span>&larr; <a class="prev" href="SatelliteBigData@version=27.html" title="Version 27">Previous Version</a></span></li><li><a href="SatelliteBigData.html" title="View latest version">View Latest Version</a></li><li><span><a class="next" href="SatelliteBigData@version=29.html" title="Version 29">Next Version</a> &rarr;</span></li><li class="last"><a href="SatelliteBigData@action=diff&amp;version=28.html">Last Change</a></li>
          </ul>
        <hr />
      </div>
    <div id="content" class="wiki">
        <table id="info" summary="Revision info">
          <tbody>
            <tr><th scope="row">
                Version 28 (modified by severin, <a class="timeline" href="../timeline@from=2009-03-19T18%253A10%253A23+0900&amp;precision=second.html" title="2009-03-19T18:10:23+0900 in Timeline">16 years</a> ago)
            </th></tr>
            <tr><td class="message">
              <p>
--
</p>

            </td></tr>
          </tbody>
        </table>
      <div class="wikipage searchable">
        
          <h1 id="Satellitemeetingforhandlingnext-generationdatasets">Satellite meeting for handling next-generation datasets</h1>
<h2 id="Topics">Topics</h2>
<p>
As more and more projects are spewing out datasets we might have to look at new paradigms of working with them. The conventional way of storing everything in a database and querying with SQL has become challenging or just impossible. Data are now more often stored in their original data formats.
</p>
<p>
The object of this meeting is to exchange ideas on this topic and discuss possible solutions or practices.
</p>
<p>
Topics:
</p>
<ul><li>Next-gen sequencing
</li><li>Protein-interacting networks?
</li><li>Cloud storage/compute?
</li></ul><h2 id="Chairperson">Chairperson</h2>
<ul><li>Jan Aerts
</li><li> Jessica Severin
</li></ul><h2 id="Date">Date</h2>
<ul><li>18th March, 2009
</li></ul><h2 id="Room">Room</h2>
<ul><li>Meeting Room (1F)
</li></ul><h2 id="Attendees">Attendees</h2>
<ul><li>Jan Aerts
</li><li>Jessica Severin
</li><li>José María Fernández
</li><li>Todd Harris
</li><li>Yunsun Nam
</li><li>Pierre Lindenbaum
</li><li>Keun-Joon Park
</li><li>Raoul Jean Pierre Bonnal
</li><li>Yasukazu "yaskaz" Nakamura (yaskaz@…)
</li></ul><h2 id="Notes">Notes</h2>
<p>
Data storage will not be a problem (if you have enogh money). But downloading and/or manipulating Big sequence data must be problem for data producers (Sanger, RIKEN OSC), data distribution providers (NCBI, SRA, DDBJ) and end users after the data is made public. Is there a possible solution? 
</p>
<p>
And I have a big question: Do you really need and/or use "Short Read Archive?" -- yaskaz
</p>
<ul><li>yes, there is interesting biology in there and comparing our new experiments against existing knowledge is needed - jessica 
</li></ul><p>
                                               
                                             
</p>
<h2 id="Discussion">Discussion</h2>
<p>
The purpose of this workshop was to discuss the approaches people use to handle their data. We could obviously not go into the very technical approaches as are being applied to very large datasets such as the Short Read Archive because that requires a different expertise.
</p>
<p>
Three different levels in the data handling pipeline were discussed.
</p>
<h4 id="Storage">Storage</h4>
<p>
We listed the different types of storage solution used in different fields according to the amount and type of data. A distinction was made between storing a lot of small objects versus storing a few huge objects. Issues like latency and variety of data influence this distinction as well.
</p>
<p>
Protein-protein interaction datasets typically consist of a relatively small number of small objects. This type of data requires no advanced storage systems or tweaks to common systems; a simple RDBMS will do.
Data like genome sequences and assembly data involve larger objects (1 genome sequence -&gt; 3Gb), but are still easily manageable on a standard filesystem.
Really big objects such as the data from simulations [IS THIS CORRECT?] require specialized storage systems such as ZFS, lustre or PVFS.
</p>
<p>
In contrast to the above, diffraction results, microarray results or next-gen sequencing reads involve a largish number of objects which become more difficult to query. They are typically still stored in RDBMS but might require some tweaking that digresses from a normalized relational database model. Apart from obvious things to do such as creating good indices, further optimization can be found by using as few joins as possible and therefore organizing the data so that it can be stored in 2 or 3 tables (e.g. eeDB).
</p>
<p>
Several attendees are looking into new ways to store their data because they are hitting the ceiling of their storage capacity. Several technologies were mentioned, including OGSADAI which is a grid-based solution where you can setup several databases that then can be queried as one. Other technologies like the cloud might also provide part of a solution. Amazon S3 and <a class="missing wiki">GoogleBase?</a> allow for storing very large amounts of data (large numbers of large objects) and are relatively cheap. The upload is however very slow to these systems which must be taken into account. In addition, these services are commercial and it might be dangerous to store non-public data there. A possible solution mentioned involves creating an encrypted data image to be uploaded instead of the original dataset. Another issue is that these companies might in the future decide to increase their prices or even stop their activities. Using the cloud for data storage therefore means that you still need a local backup in case this happens.
</p>
<p>
[Jessica: can you add your vertical red line to this as well?]
</p>
<h4 id="Querying">Querying</h4>
<p>
Several approaches to make the data accessible to others were discussed. For smaller datasets regular SQL can be used to get to the data (example: the Ensembl MySQL server on ensembldb.ensembl.org). If the database schema has to be tweaked to allow larger datasets (e.g. using a minimal number of tables such as eeDB) an API becomes necessary. A RESTful API where an object can be retrieved by URL was mentioned as useful in this case. For even larger datasets you want to try and limit the types of queries people can run. This way you can build a toolkit to ask this limited number of questions, but optimize that toolkit so that latency is reduced.
</p>
<p>
Streaming
</p>
<blockquote>
<p>
SQL -&gt; API -&gt; toolkit
</p>
<blockquote>
<p>
<sup>      |
|      |
+------+
</sup></p>
</blockquote>
</blockquote>
<h4 id="Processing">Processing</h4>
<p>
Apart from storage and getting the data out again we briefly discussed how to actually work with these data, i.e. running scripts on them. It is clear that shipment of large datasets should be avoided; it is often much simpler to bring the software to the data instead.
</p>
<p>
The main message from this discussion was that each database should be optimized for its own data and will therefore be completely different from the next. However, it is the API that can provide interoperability if we can come up with a common set of terms for querying. We believe that current web-technologies like mashups based on very simple APIs like Flickr's can be an alternative to other grid-based approached for integrating data.
</p>
<h2 id="Workgoupdiscussiontopics">Work goup discussion topics</h2>
<p>
next gen sequencer WG (big data and seq analysis)
</p>
<p>
applications and use cases:
</p>
<ul><li>all processing must be run on a server, can not be run on a laptop, how to control and monitor such processes from something like a laptop
</li><li>local vs webservice. web service convenient but maybe not so secure which means people need to install our systems locally. but also collaboration needs access to data and tools
</li><li>how to define the quality of sequence?  contamination, auto screen tool?
</li><li>moving to a data-driven research, not so much hypothesis driven. how to FIND biology in big datasets?
</li><li>searching / comparing now mostly happening on the metadata (time series, tissue, environments)
</li><li>metagenomics or novel genomes, no reference, can our tools do this?  How?
</li><li>maybe we get a strange sequence out of our instrument, how do we know?  is it error? is it interesting biology?  CG content off, maybe technical errors (concatonated primers)
</li><li>how to bring the computation to the data because bring the data to the computation (1TB streaming) maybe not best approach
</li><li>as sequencer companies build more and more sequence processing into the instruments, more and more processing becomes routine. it is a moving target. how do we aid in new biological discovery?  moving beyond simple processing and simple work flows.
</li></ul><p>
what is big data?
</p>
<ul><li>read level, contigs, assembly, scaffold each different level
</li><li>amount vs complexity!!  maybe amount not so hard, but complexity ....
</li><li>variation, snp
</li><li>data is big when the applications can not handle it.  If 1GB of data crashes a viewer program than that is BIG.  If 100k elements takes a day to upload, this upload must happen every time we want to work with the data, and it takes another day to filter for a service, then that is probably BIG data. 
</li></ul><p>
How to collaborate and communicate?
</p>
<ul><li>emailing files or providing files on secure FTP sites works for now
</li><li>reducing dataset complexity by processing can make BIG data small, but does this throw away biology?
</li><li>secure webservices are maybe a much better solution where more advanced tools can be built which can work with sections of the dataset, but this requires the ability to provide sub-access queries into these datasets
</li></ul><p>
Data production centers
</p>
<ul><li>RIKEN OSC-LSA <a class="ext-link" href="http://www.osc.riken.jp/"><span class="icon"> </span>http://www.osc.riken.jp/</a> is producing lots of data, but this data must be managed, manipulated, and mined for biology before it can be released.  EdgeExpressDB (eeDB) was developed during FANTOM4 project and is now being used for in-house big data management and visualization of big datasets.  eeDB is effectively an object-database which is implemented as an API and webservices. The system will be ported to C and file indexes this summer which will give us at least a 100x performance boost.  Currently this API toolkit and webservices are written in perl with a narrow/deep mysql snowflake schema. This generation1 system of the API can manipulate short-read data for our internal research purposes and is proving to scale very well. eeDB works with node and network, sequence tag, mapping, and expression data at the level of billions of elements very easily.  Queries can access individual objects, edges, and work with streams or sets of objects queried by regions, node, or networks. 
</li><li>SRA is still evaluating technology to do region based access of short-reads
</li></ul><p>
Working with existing big data
</p>
<ul><li>SRA, GEO, <a class="missing wiki">ArrayExpress?</a>
</li><li>now most of us pull the whole thing down and then work with it
</li><li>sometimes it is even hard to send for submission, sometimes DVD to move it around. not optimal.
</li><li>what are the queries we want to do?
<ul><li>what are the reads in this region, definitely 
</li></ul></li><li>but also SRA maybe does not want do everything, but they will not turn data down and want everyone to send them the published data
</li><li>maybe not all data will end up in ONE archive (because it is so big). maybe need to query multiple centers to find all data (DDBJ, SRA, GEO, <a class="missing wiki">ArrayExpress?</a>, korea? China?)
</li></ul><p>
What are the common ways we would want to query?
</p>
<ul><li>should we try to define? yes
</li><li>who should or can provide the data?
<ul><li>maybe the the data production centers can also provide access to data
</li><li>eventually the <a class="missing wiki">DataCenters?</a> will also provide fine-grained access
</li><li>the new high data production rates are going to make it expensive for central data providers
</li></ul></li><li>region query again we need! means mapping needed by the services
</li><li>but also want the read Sequence since we can extract SNP, need quality
</li><li>NEED data providers to create complete meta-data descriptions.
</li></ul><p>
IN THE END the goal is to find biology. Having access to the individual data elements is critical and this can not be just locked away inside files that can not be internally accessed.
</p>
<h2 id="TODOs">TODOs</h2>
<ul><li>Design tests for BioSQL on different scenario related to <strong><a class="missing wiki">BigData?</a></strong> BigD
<ul><li>Can BioSQL be used for short reads?
</li></ul></li><li>Test BioSQL performances on BigD
</li></ul>
        
        
      </div>
          <h3>Attachments</h3>
          <ul>
              <li>
      <a href="../attachment/wiki/SatelliteBigData/handling_data.pdf.html" title="View attachment">handling_data.pdf</a>
      <a href="../raw-attachment/wiki/SatelliteBigData/handling_data.pdf" title="Download" class="trac-rawlink"><img src="../chrome/common/download.png" alt="Download" /></a>
      (<span title="23553 bytes">23.0 KB</span>) - added by <em>jan.aerts</em>
      <a class="timeline" href="../timeline@from=2009-03-20T16%253A23%253A50+0900&amp;precision=second.html" title="2009-03-20T16:23:50+0900 in Timeline">16 years</a> ago.
              </li>
          </ul>
    </div>
    <div id="altlinks">
      <h3>Download in other formats:</h3>
      <ul>
        <li class="last first">
          <a rel="nofollow" href="http://hackathon2.dbcls.jp/wiki/SatelliteBigData?version=28&amp;format=txt">Plain Text</a>
        </li>
      </ul>
    </div>
    </div>
    <div id="footer" lang="en" xml:lang="en"><hr />
      <a id="tracpowered" href="http://trac.edgewall.org/"><img src="../chrome/common/trac_logo_mini.png" height="30" width="107" alt="Trac Powered" /></a>
      <p class="left">
        Powered by <a href="../about.html"><strong>Trac 0.11.6</strong></a><br />
        By <a href="http://www.edgewall.org/">Edgewall Software</a>.
      </p>
      <p class="right">Visit the Trac open source project at<br /><a href="http://trac.edgewall.org/">http://trac.edgewall.org/</a></p>
    </div>
  </body>
</html>