<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

    <html xmlns="http://www.w3.org/1999/xhtml">
  
  


  

  <head>
    <title>
      SatelliteSeqAnalysis (diff) – BioHackathon 2009
    </title>
        <link rel="search" href="../search.1.html" />
        <link rel="help" href="TracGuide.html" />
        <link rel="up" href="SatelliteSeqAnalysis@action=history.html" title="Page history" />
        <link rel="next" href="SatelliteSeqAnalysis@action=diff&amp;version=26.html" title="Version 26" />
        <link rel="start" href="../wiki.1.html" />
        <link rel="stylesheet" href="../chrome/common/css/trac.css" type="text/css" /><link rel="stylesheet" href="../chrome/common/css/wiki.css" type="text/css" /><link rel="stylesheet" href="../chrome/common/css/diff.css" type="text/css" />
        <link rel="prev" href="SatelliteSeqAnalysis@action=diff&amp;version=24.html" title="Version 24" />
        <link rel="shortcut icon" href="../chrome/common/trac.ico" type="image/x-icon" />
        <link rel="icon" href="../chrome/common/trac.ico" type="image/x-icon" />
      <link type="application/opensearchdescription+xml" rel="search" href="../search/opensearch" title="Search BioHackathon 2009" />
    <script type="text/javascript" src="../chrome/common/js/jquery.js"></script><script type="text/javascript" src="../chrome/common/js/trac.js"></script><script type="text/javascript" src="../chrome/common/js/search.js"></script><script type="text/javascript" src="http://hackathon2.dbcls.jp/chrome/common/js/diff.js"></script>
    <!--[if lt IE 7]>
    <script type="text/javascript" src="/chrome/common/js/ie_pre7_hacks.js"></script>
    <![endif]-->
    <link rel="stylesheet" type="text/css" href="../chrome/common/css/diff.css" /><meta name="ROBOTS" content="NOINDEX, NOFOLLOW" />
  </head>
  <body>
    <div id="banner">
      <div id="header">
        <a id="logo" href="../index.html"><img src="../chrome/site/bh2009-logo.jpg" alt="" /></a>
      </div>
      <form id="search" action="../search.1.html" method="get">
        <div>
          <label for="proj-search">Search:</label>
          <input type="text" id="proj-search" name="q" size="18" value="" />
          <input type="submit" value="Search" />
        </div>
      </form>
      <div id="metanav" class="nav">
    <ul>
      <li class="first"><a href="http://hackathon2.dbcls.jp/login">Login</a></li><li><a href="TracGuide.html">Help/Guide</a></li><li><a href="../about.html">About Trac</a></li><li class="last"><a href="../prefs.html">Preferences</a></li>
    </ul>
  </div>
    </div>
    <div id="mainnav" class="nav">
    <ul>
      <li class="first active"><a href="../wiki.1.html">Wiki</a></li><li><a href="../timeline.html">Timeline</a></li><li><a href="../milestone/index.html">Roadmap</a></li><li><a href="../browser.html">Browse Source</a></li><li><a href="../report.html">View Tickets</a></li><li class="last"><a href="../search.1.html">Search</a></li>
    </ul>
  </div>
    <div id="main">
      <div id="ctxtnav" class="nav">
        <h2>Context Navigation</h2>
          <ul>
              <li class="first"><span>&larr; <a class="prev" href="SatelliteSeqAnalysis@action=diff&amp;version=24.html" title="Version 24">Previous Change</a></span></li><li><a href="SatelliteSeqAnalysis@action=history.html" title="Page history">Wiki History</a></li><li class="last"><span><a class="next" href="SatelliteSeqAnalysis@action=diff&amp;version=26.html" title="Version 26">Next Change</a> &rarr;</span></li>
          </ul>
        <hr />
      </div>
    <div id="content" class="wiki">
      <h1>Changes
        between
          <a href="SatelliteSeqAnalysis@version=24.html">Version 24</a> and
        <a href="SatelliteSeqAnalysis@version=25.html">Version 25</a> of
        <a href="SatelliteSeqAnalysis@version=25.html">SatelliteSeqAnalysis</a>
      </h1>
      <form method="post" id="prefs" action="SatelliteSeqAnalysis@version=25.html"><div><input type="hidden" name="__FORM_TOKEN" value="2c93bff27ce469855e39283a" /></div>
        <div>
          <input type="hidden" name="action" value="diff" />
          <input type="hidden" name="version" value="25" />
          <input type="hidden" name="old_version" value="24" />
    <label for="style">View differences</label>
    <select id="style" name="style">
      <option selected="selected" value="inline">inline</option>
      <option value="sidebyside">side by side</option>
    </select>
    <div class="field">
      Show <input type="text" name="contextlines" id="contextlines" size="2" maxlength="3" value="2" />
      <label for="contextlines">lines around each change</label>
    </div>
    <fieldset id="ignore">
      <legend>Ignore:</legend>
      <div class="field">
        <input type="checkbox" id="ignoreblanklines" name="ignoreblanklines" />
        <label for="ignoreblanklines">Blank lines</label>
      </div>
      <div class="field">
        <input type="checkbox" id="ignorecase" name="ignorecase" />
        <label for="ignorecase">Case changes</label>
      </div>
      <div class="field">
        <input type="checkbox" id="ignorewhitespace" name="ignorewhitespace" />
        <label for="ignorewhitespace">White space changes</label>
      </div>
    </fieldset>
    <div class="buttons">
      <input type="submit" name="update" value="Update" />
    </div>
        </div>
      </form>
      <dl id="overview">
        <dt class="property time">Timestamp:</dt>
        <dd class="time">
          2009/03/24 18:14:50 (<a class="timeline" href="../timeline@from=2009-03-24T18%253A14%253A50+0900&amp;precision=second.html" title="2009-03-24T18:14:50+0900 in Timeline">16 years</a> ago)
        </dd>
        <dt class="property author">Author:</dt>
        <dd class="author">
          mn
        </dd>
        <dt class="property message">Comment:</dt>
        <dd class="message">
          
          <p>
--
</p>

        </dd>
      </dl>
      <div class="diff">
        <div class="legend" id="diff-legend">
          <h3>Legend:</h3>
          <dl>
            <dt class="unmod"></dt><dd>Unmodified</dd>
            <dt class="add"></dt><dd>Added</dd>
            <dt class="rem"></dt><dd>Removed</dd>
            <dt class="mod"></dt><dd>Modified</dd>
          </dl>
        </div>
        <div class="diff">
  <ul class="entries">
      <li class="entry">
        <h2 id="file0">
          <a href="SatelliteSeqAnalysis@version=25.html">SatelliteSeqAnalysis</a>
        </h2>
        <table class="inline" summary="Differences" cellspacing="0">
              <colgroup><col class="lineno" /><col class="lineno" /><col class="content" /></colgroup>
              <thead>
                <tr>
                  <th title="Version 24">
                    <a href="SatelliteSeqAnalysis@version=24.html#L8">
                      v24</a>
                  </th>
                  <th title="Version 25">
                    <a href="SatelliteSeqAnalysis@version=25.html#L8">
                      v25</a>
                  </th>
                  <th> </th>
                </tr>
              </thead>
            <tbody class="unmod">
                  <tr>
                          <th>8</th><th>8</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>9</th><th>9</th><td class="l"><span>== Mapping tools ==</span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>10</th><th> </th><td class="l"><span>Maq[[BR]]</span> </td>
                      </tr><tr>
                        <th>11</th><th> </th><td class="l"><span>ELAND[[BR]]</span> </td>
                      </tr><tr>
                        <th>12</th><th> </th><td class="l"><span>[http://bowtie-bio.sourceforge.net/index.shtml BOWTIE] Ultra-short reads, Burrows-Wheeler index. [[BR]]</span> </td>
                      </tr><tr>
                        <th>13</th><th> </th><td class="l"><span>[http://www.dnastar.com/ Lasergene] Support 454/Solexa. GUI interface for displaying alignments and contigs organization (scaffolds) [[BR]]</span> </td>
                      </tr><tr>
                        <th>14</th><th> </th><td class="l"><span><del></del>[http://www.454.com/products-solutions/analysis-tools/gs-reference-mapper.asp GSMapper]</span> </td>
                      </tr>
                      <tr>
                        <th> </th><th>10</th><td class="r"><span>&nbsp;* Maq</span> </td>
                      </tr><tr>
                        <th> </th><th>11</th><td class="r"><span>&nbsp;* ELAND</span> </td>
                      </tr><tr>
                        <th> </th><th>12</th><td class="r"><span>&nbsp;* [http://bowtie-bio.sourceforge.net/index.shtml BOWTIE] Ultra-short reads, Burrows-Wheeler index.</span> </td>
                      </tr><tr>
                        <th> </th><th>13</th><td class="r"><span>&nbsp;* [http://www.dnastar.com/ Lasergene] Support 454/Solexa. GUI interface for displaying alignments and contigs organization (scaffolds) </span> </td>
                      </tr><tr class="last">
                        <th> </th><th>14</th><td class="r"><span><ins>&nbsp;* </ins>[http://www.454.com/products-solutions/analysis-tools/gs-reference-mapper.asp GSMapper]</span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>15</th><th>15</th><td class="l"><span>&nbsp; &nbsp; *&nbsp; Map reads to any reference genome and generate a consensus sequence.</span> </td>
                  </tr><tr>
                          <th>16</th><th>16</th><td class="l"><span>&nbsp; &nbsp; * Easily view all differences compared to the reference sequence with automatic output to separate files: Insertions (blocks up to 50 bases), Deletions (blocks up to 50 bases), SNPs</span> </td>
                  </tr>
            </tbody>
              <tbody class="skipped">
                <tr>
                  <th><a href="SatelliteSeqAnalysis@version=24.html#L18">…</a></th>
                  <th><a href="SatelliteSeqAnalysis@version=25.html#L18">…</a></th>
                  <td> </td>
                </tr>
              </tbody>
            <tbody class="unmod">
                  <tr>
                          <th>18</th><th>18</th><td class="l"><span>&nbsp; &nbsp; * Compare large, complex genomes of any size including: resequencing of whole genomes from humans, plants, yeasts, bacteria, fungi, viruses, YACs, BACs, fosmids</span> </td>
                  </tr><tr>
                          <th>19</th><th>19</th><td class="l"><span>&nbsp; &nbsp; * Data outputs: fna.file (sequence of contigs, FASTA format), qual.file (corresponding Phred equivalent quality score), ace.file (consensus alignment of the reads against a given reference sequence)</span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>20</th><th> </th><td class="l"><span>[[BR]]</span> </td>
                      </tr><tr>
                        <th>21</th><th> </th><td class="l"><span>[http://compbio.cs.toronto.edu/shrimp/ SHRiMP] supports letter-space (454/Solexa), color-space (Solid) and Helicos 2-pass space. Spaced-seeds followed by S-W. [[BR]]</span> </td>
                      </tr><tr>
                        <th>22</th><th> </th><td class="l"><span>[http://bx.psu.edu/miller_lab/dist/README.lastz-1.01.50/README.lastz-1.01.50.html Lastz] Blast(z) variant. Has parameter settings tuned for 454 and Solexa mapping for both reference based assembly and variant discovery. [[BR]]</span> </td>
                      </tr><tr>
                        <th>23</th><th> </th><td class="l"><span>[http://emboss.sourceforge.net/ EMBOSS]&nbsp; [[BR]]</span> </td>
                      </tr>
                      <tr>
                        <th> </th><th>20</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>21</th><td class="r"><span>&nbsp;* [http://compbio.cs.toronto.edu/shrimp/ SHRiMP] supports letter-space (454/Solexa), color-space (Solid) and Helicos 2-pass space. Spaced-seeds followed by S-W. </span> </td>
                      </tr><tr>
                        <th> </th><th>22</th><td class="r"><span>&nbsp;* [http://bx.psu.edu/miller_lab/dist/README.lastz-1.01.50/README.lastz-1.01.50.html Lastz] Blast(z) variant. Has parameter settings tuned for 454 and Solexa mapping for both reference based assembly and variant discovery. </span> </td>
                      </tr><tr>
                        <th> </th><th>23</th><td class="r"><span>&nbsp;* [http://emboss.sourceforge.net/ EMBOSS]</span> </td>
                      </tr><tr class="last">
                        <th> </th><th>24</th><td class="r"><span>&nbsp;* [http://last.cbrc.jp/ LAST]</span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>24</th><th>25</th><td class="l"><span>many mores.... but never used.</span> </td>
                  </tr><tr>
                          <th>25</th><th>26</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>26</th><th>27</th><td class="l"><span>== Assemble tools ==</span> </td>
                  </tr><tr>
                          <th>27</th><th>28</th><td class="l"><span></span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>28</th><th> </th><td class="l"><span><del></del>[http://www.454.com/products-solutions/analysis-tools/gs-de-novo-assembler.asp GSAssembly] </span> </td>
                      </tr>
                      <tr class="last">
                        <th> </th><th>29</th><td class="r"><span><ins>&nbsp;* </ins>[http://www.454.com/products-solutions/analysis-tools/gs-de-novo-assembler.asp GSAssembly] </span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>29</th><th>30</th><td class="l"><span>&nbsp; &nbsp; *&nbsp; Performs whole genome shotgun assembly of genomes with or without paired-end data.</span> </td>
                  </tr><tr>
                          <th>30</th><th>31</th><td class="l"><span>&nbsp; &nbsp; * Order contigs into scaffolds using supported paired-end reads.</span> </td>
                  </tr>
            </tbody>
              <tbody class="skipped">
                <tr>
                  <th><a href="SatelliteSeqAnalysis@version=24.html#L33">…</a></th>
                  <th><a href="SatelliteSeqAnalysis@version=25.html#L34">…</a></th>
                  <td> </td>
                </tr>
              </tbody>
            <tbody class="unmod">
                  <tr>
                          <th>33</th><th>34</th><td class="l"><span>&nbsp; &nbsp; * Select the read files you want to assemble by browsing in the GUI.</span> </td>
                  </tr><tr>
                          <th>34</th><th>35</th><td class="l"><span>&nbsp; &nbsp; * Data outputs: fna.file (sequence of contigs, FASTA format), qual.file (corresponding Phred equivalent quality score), ace.file (alignment of the reads to contig sequence)</span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>35</th><th> </th><td class="l"><span>[[BR]]</span> </td>
                      </tr><tr>
                        <th>36</th><th> </th><td class="l"><span>[http://www.dnastar.com/ Lasergene] Support 454/Solexa. GUI interface for displaying alignments and contigs organization (scaffolds) [[BR]]</span> </td>
                      </tr><tr>
                        <th>37</th><th> </th><td class="l"><span><del>[http://www.ebi.ac.uk/~zerbino/velvet/ Velvet][[BR]</del>]</span> </td>
                      </tr><tr>
                        <th>38</th><th> </th><td class="l"><span><del>[http://emboss.sourceforge.net/ EMBOSS] [[BR]</del>]</span> </td>
                      </tr>
                      <tr>
                        <th> </th><th>36</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>37</th><td class="r"><span>&nbsp;* [http://www.dnastar.com/ Lasergene] Support 454/Solexa. GUI interface for displaying alignments and contigs organization (scaffolds)</span> </td>
                      </tr><tr>
                        <th> </th><th>38</th><td class="r"><span><ins>&nbsp;* [http://www.ebi.ac.uk/~zerbino/velvet/ Velvet</ins>]</span> </td>
                      </tr><tr class="last">
                        <th> </th><th>39</th><td class="r"><span><ins>&nbsp;* [http://emboss.sourceforge.net/ EMBOSS</ins>]</span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>39</th><th>40</th><td class="l"><span>many mores.... but never used.</span> </td>
                  </tr><tr>
                          <th>40</th><th>41</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>41</th><th>42</th><td class="l"><span>== Platforms == </span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>42</th><th> </th><td class="l"><span><del>[http://www.454.com/ Roche, 454] [[BR]</del>]</span> </td>
                      </tr><tr>
                        <th>43</th><th> </th><td class="l"><span><del>[http://www.illumina.com/pages.ilmn?ID=203 Illumina, Solexa] [[BR]</del>]</span> </td>
                      </tr><tr>
                        <th>44</th><th> </th><td class="l"><span><del>[http://www.appliedbiosystems.co.jp/website/jp/home/index.jsp AB, Solid] [[BR]</del>]</span> </td>
                      </tr>
                      <tr>
                        <th> </th><th>43</th><td class="r"><span><ins>&nbsp;* [http://www.454.com/ Roche, 454</ins>]</span> </td>
                      </tr><tr>
                        <th> </th><th>44</th><td class="r"><span><ins>&nbsp;* [http://www.illumina.com/pages.ilmn?ID=203 Illumina, Solexa</ins>]</span> </td>
                      </tr><tr class="last">
                        <th> </th><th>45</th><td class="r"><span><ins>&nbsp;* [http://www.appliedbiosystems.co.jp/website/jp/home/index.jsp AB, Solid</ins>]</span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>45</th><th>46</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>46</th><th>47</th><td class="l"><span>== Notes ==</span> </td>
                  </tr><tr>
                          <th>47</th><th>48</th><td class="l"><span></span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>48</th><th> </th><td class="l"><span>Every new software must handle this "huge" amount of data. [[BR]]</span> </td>
                      </tr><tr>
                        <th>49</th><th> </th><td class="l"><span>Some veriations SNP, deletion, insertion can by identified during mapping/assembly process, other things more "new biology" needs more work by bionformaticians.&nbsp; [[BR]]</span> </td>
                      </tr><tr>
                        <th>50</th><th> </th><td class="l"><span><del></del>Support from biologists is required in order to discover more new things.</span> </td>
                      </tr>
                      <tr>
                        <th> </th><th>49</th><td class="r"><span>&nbsp;* Every new software must handle this "huge" amount of data. </span> </td>
                      </tr><tr>
                        <th> </th><th>50</th><td class="r"><span>&nbsp;* Some veriations SNP, deletion, insertion can by identified during mapping/assembly process, other things more "new biology" needs more work by bionformaticians.&nbsp; </span> </td>
                      </tr><tr class="last">
                        <th> </th><th>51</th><td class="r"><span><ins>&nbsp;* </ins>Support from biologists is required in order to discover more new things.</span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>51</th><th>52</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>52</th><th>53</th><td class="l"><span>== TODOs ==</span> </td>
                  </tr><tr>
                          <th>53</th><th>54</th><td class="l"><span></span> </td>
                  </tr>
            </tbody><tbody class="mod">
                      <tr class="first">
                        <th>54</th><th> </th><td class="l"><span>Connected to SatelliteBigData a LIMS infrastructure is suggested for tracking history and infer problems coming from wet-lab. [[BR]]</span> </td>
                      </tr><tr>
                        <th>55</th><th> </th><td class="l"><span>From SatelliteVisualization these tools could be very usefull to explore at any level all the informations collected, from the alignment (lower) up to annotation/comparison (higher). The informations are intended as local (faltfile/databases/webservices) or remote (databases/webservices).&nbsp; [[BR]]</span> </td>
                      </tr><tr>
                        <th>56</th><th> </th><td class="l"><span>Editing in real time, providing alternatives for hypothesis. [[BR]]</span> </td>
                      </tr>
                      <tr>
                        <th> </th><th>55</th><td class="r"><span>Connected to SatelliteBigData a LIMS infrastructure is suggested for tracking history and infer problems coming from wet-lab. </span> </td>
                      </tr><tr>
                        <th> </th><th>56</th><td class="r"><span></span> </td>
                      </tr><tr>
                        <th> </th><th>57</th><td class="r"><span>From SatelliteVisualization these tools could be very usefull to explore at any level all the informations collected, from the alignment (lower) up to annotation/comparison (higher). The informations are intended as local (faltfile/databases/webservices) or remote (databases/webservices).&nbsp; </span> </td>
                      </tr><tr>
                        <th> </th><th>58</th><td class="r"><span></span> </td>
                      </tr><tr class="last">
                        <th> </th><th>59</th><td class="r"><span>Editing in real time, providing alternatives for hypothesis. </span> </td>
                      </tr>
            </tbody><tbody class="unmod">
                  <tr>
                          <th>57</th><th>60</th><td class="l"><span></span> </td>
                  </tr><tr>
                          <th>58</th><th>61</th><td class="l"><span>Easy tools to integrate informations coming from different platforms (illumina, 454, solid).</span> </td>
                  </tr>
            </tbody>
        </table>
      </li>
  </ul>
</div>
      </div>
  </div>
    </div>
    <div id="footer" lang="en" xml:lang="en"><hr />
      <a id="tracpowered" href="http://trac.edgewall.org/"><img src="../chrome/common/trac_logo_mini.png" height="30" width="107" alt="Trac Powered" /></a>
      <p class="left">
        Powered by <a href="../about.html"><strong>Trac 0.11.6</strong></a><br />
        By <a href="http://www.edgewall.org/">Edgewall Software</a>.
      </p>
      <p class="right">Visit the Trac open source project at<br /><a href="http://trac.edgewall.org/">http://trac.edgewall.org/</a></p>
    </div>
  </body>
</html>
